src/box/index.cc

```diff
@@ -740,22 +740,26 @@ iterator_position_unpack(const char *packed_pos, const char *packed_pos_end,
 int
 iterator_position_validate(const char *pos, uint32_t pos_part_count,
 			   const char *key, uint32_t key_part_count,
-			   struct key_def *cmp_def, enum iterator_type type)
+			   struct index_def *ind_def, enum iterator_type type)
 {
 	int cmp;
+	struct key_def *cmp_def = ind_def->cmp_def;
 	/* Position must be compatible with the index. */
 	if (cmp_def->part_count != pos_part_count)
 		goto fail;
 	const char *pos_end;
 	if (key_validate_parts(cmp_def, pos, pos_part_count, true, &pos_end) != 0)
 		goto fail;
-	/* Position msut meet the search criteria. */
-	cmp = key_compare(pos, pos_part_count, HINT_NONE,
-			  key, key_part_count, HINT_NONE, cmp_def);
-	if (iterator_direction(type) * cmp < 0)
-		goto fail;
-	if ((type == ITER_EQ || type == ITER_REQ) && cmp != 0)
-		goto fail;
+	/* Can skip, if it is rtree index. */
+	if (ind_def->type == TREE) {
+		/* Position msut meet the search criteria. */
```

Comment:
Please, fix the typo: `msut` -> `must`. Yes, it was here before you, let's just fix it along the way.